src/utils/events/tauri-shortcuts.ts

```diff
@@ -177,6 +178,23 @@ export async function setupTauriEvents() {
     }
   })
 
+  const fileUrlHandler = async (urls: string[] | null) => {
+    if (!urls) {
+      return
+    }
+    try {
+      for (const url of urls) {
+        const file = await accessReadFile(url.replace('file://', ''))
+        openTab(file)
+      }
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  await onOpenUrl(fileUrlHandler)
```

Comment:
If this happens concurrently to access manager, could there be problems? Tauri JS permissions are super restricted, so I tried to make all file reading go through access manager now, that has some record keeping and native permissions.

I guess if you "Open With..." Tauri will make sure JS has permissions to read the file. It would just be nice if AccessManager reached out to this callback on its own (via a `listen`?)

Just a feeling instead of an actual functionality concern. You said you tested this on Mac, so I think that's a good sign.